api/v1alpha1/basic_auth_types.go

```diff
@@ -0,0 +1,19 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// BasicAuth defines the configuration for the HTTP Basic Authentication.
+type BasicAuth struct {
+	// Username-hashed password pairs used to verify user credentials in the
+	// "Authorization" header.
+	//
+	// The value needs to be the htpasswd format, for example: "user1:{SHA}hashed_user1_password".
+	// Right now, ony SHA hash algorithm is supported.
+	// Reference to https://httpd.apache.org/docs/2.4/programs/htpasswd.html
+	//
+	// +kubebuilder:validation:MinItems=1
+	Users []string `json:"users"`
```

Comment:
must be a secret
we can either support multiple basic auth secrets https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret which might be too much to type out and manage or support a generic secret with `data` containing the key `htpasswd` with its contents in it

Reply:
I vote to put `htpaswd` file in the `data` of a generic secret,  like the nginx k8s controller does: https://kubernetes.github.io/ingress-nginx/examples/auth/basic/

Reply:
yeah sgtm, reg key naming, should it be `.htpasswd` to reflect the name of the file ?

Reply:
That's a good point.